shell/common/display.h

```diff
@@ -0,0 +1,47 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_COMMON_DISPLAY_H_
+#define FLUTTER_SHELL_COMMON_DISPLAY_H_
+
+namespace flutter {
+
+/// Unique IO per display that is stable until the Flutter application restarts.
+/// See also: `flutter::Display`
+typedef uint64_t DisplayId;
+
+/// Some devices do not support multiple displays, in such cases the embedding
+/// might not provide a device identifier. This value is used as the device id
+/// in such cases. There must only be one screen in such cases.
+static constexpr DisplayId kSingleScreenDeviceId = 0;
```

Comment:
So, shouldn't devices that have multiple displays be prohibited from using this value, so that they can be distinguished?

What if the embedding wants to use a platform-supplied ID that could be zero?

Reply:
Good point. I wanted to have a reserved value for platforms that only support a single embedder. I can fix the doc here and in the `embedder.h` file to indicate the reserved nature of this value.

Do you think `0` is a bad choice for this?

Reply:
I think that the concept of it being "unknown" shouldn't be conflated with the id, since the platform might provide zero as a valid ID screen ID (it's pretty likely, actually), and then the embedding would have to do some acrobatics to keep from using zero.

Maybe there should just be another way to find out if this is a single display device, and this just isn't needed?  Shouldn't the `DisplayManager` have a way to enumerate the displays, and if there's only one, then it's a single-screen device? Ideally, a user of the DisplayManager wouldn't need to have any special code to handle single-display devices verses multi-display devices, they'd just iterate over fewer devices.

Reply:
Based on our discussion, i've made some changes the ones in this commit specially: 
https://github.com/flutter/engine/pull/21355/commits/a7f95f373db32ea9f4c13c429f76729e753c5754. Hopefully that makes things a little clearer.